R/initialize_modules.R

```diff
@@ -968,16 +970,30 @@ set_param_vector <- function(field, module, module_input) {
 
   # Extract the value of the parameter vector
   field_value <- module_input[[field_value_name]]
+  estimation_type_value <- module_input[[field_estimation_name]]
 
   # Resize the field in the module if it has multiple values
   if (length(field_value) > 1) module[[field]]$resize(length(field_value))
 
   # Assign each value to the corresponding position in the parameter vector
+  # TODO: this is currently set up to handle a single estimation type value
+  # for all values in the parameter vector. If this is not the case, we need to
+  # modify this logic to handle multiple estimation types.
```

Comment:
Currently, this is just a limitation of the R interface right? Not a limitation of the C++ code? All we would need to do is set up a loop to set every value if the input value is greater than length 1. I support doing this if that is the case. Or documenting what would need to be changed.

Reply:
Yes I think the issue is just within the R helper functions. It probably won't take long to set up a loop and get this working.

Reply:
@Andrea-Havron-NOAA would it be appropriate to migrate the variable mapping code from the TMB example repo in this branch? Seems inline with making estimation_type different within a vector.